src/management-system-v2/components/scrollbar.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { FC, useEffect, useState, useRef } from 'react';
+
+type ScrollBarType = {
+  children: React.ReactNode;
+  width: string;
+};
+
+const ScrollBar: FC<ScrollBarType> = ({ children, width }) => {
+  const [scrollHeight, setScrollHeight] = useState(0);
+  const [scrollPosition, setScrollPosition] = useState(0);
+  const [isDragging, setIsDragging] = useState(false);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const thumbRef = useRef<HTMLDivElement>(null);
+
+  const handleScroll = () => {
+    if (containerRef.current) {
+      const { scrollTop, scrollHeight, clientHeight } = containerRef.current;
+      const newScrollHeight = (clientHeight / scrollHeight) * 100;
+      const newScrollPosition = (scrollTop / (scrollHeight - clientHeight)) * 100;
+
+      setScrollHeight(newScrollHeight);
+      setScrollPosition(newScrollPosition);
+    }
+  };
+
+  const handleMouseDown = () => {
+    setIsDragging(true);
+  };
+
+  const handleMouseUp = () => {
+    setIsDragging(false);
+  };
+
+  const handleMouseMove = (e: MouseEvent) => {
+    if (isDragging && containerRef.current && thumbRef.current) {
+      const { clientY } = e;
+      const { top, height } = containerRef.current.getBoundingClientRect();
+      const thumbHeight = thumbRef.current.clientHeight;
+
+      let newScrollTop =
+        ((clientY - top - thumbHeight / 2) / (height - thumbHeight)) *
+        containerRef.current.scrollHeight;
+
+      newScrollTop = Math.max(newScrollTop, 0);
+      newScrollTop = Math.min(
+        newScrollTop,
+        containerRef.current.scrollHeight - containerRef.current.clientHeight,
+      );
+
+      containerRef.current.scrollTop = newScrollTop;
+    }
+  };
+
+  useEffect(() => {
+    if (typeof window === 'undefined') return;
+
+    window.addEventListener('mousemove', handleMouseMove);
+    window.addEventListener('mouseup', handleMouseUp);
+
+    return () => {
+      window.removeEventListener('mousemove', handleMouseMove);
+      window.removeEventListener('mouseup', handleMouseUp);
+    };
+  }, [isDragging]);
```

Comment:
Missing dependency: handleMouseMove

Reply:
Added dependency

Reply:
I think handleMouseUp is still missing and in the next effect handleScroll is missing as a dependency.